arbiter/drivers/onedrive.cpp

```diff
@@ -0,0 +1,292 @@
+#ifndef ARBITER_IS_AMALGAMATION
+#include <arbiter/drivers/onedrive.hpp>
+#include <arbiter/arbiter.hpp>
+#include <arbiter/drivers/fs.hpp>
+#include <arbiter/util/transforms.hpp>
+#include <arbiter/util/json.hpp>
+#include <chrono>
+#include <thread>
+#endif
+
+#include <vector>
+
+#ifdef ARBITER_OPENSSL
+#include <openssl/evp.h>
+#include <openssl/err.h>
+#include <openssl/pem.h>
+#include <openssl/rsa.h>
+#include <openssl/sha.h>
+
+// See: https://www.openssl.org/docs/manmaster/man3/OPENSSL_VERSION_NUMBER.html
+#   if OPENSSL_VERSION_NUMBER >= 0x010100000
+#   define ARBITER_OPENSSL_ATLEAST_1_1
+#   endif
+
+#endif
+
+#ifdef ARBITER_CUSTOM_NAMESPACE
+namespace ARBITER_CUSTOM_NAMESPACE
+{
+#endif
+
+namespace arbiter
+{
+
+using namespace internal;
+
+namespace {
+
+static std::string getQueries(const std::string url)
+{
+    json result;
+
+    //find position of queries in url
+    std::string::size_type pos(url.find("?"));
+    if (pos == std::string::npos)
+    {
+        return result.dump();
+    }
+
+    std::string queries(url.substr(pos + 1));
+
+    do
+    {
+        //get positional values
+        //find next query
+        std::string::size_type nextQueryPos(queries.find_first_of("&"));
+        //if it doesn't exist, go to the end of the string
+        if (nextQueryPos == std::string::npos)
+            nextQueryPos = queries.size();
+        const std::string::size_type separator(queries.find("="));
+
+        //create key-value pair for json
+        const std::string key(queries.substr(0, separator));
+        const std::string value(queries.substr(separator + 1, nextQueryPos - (separator + 1)));
+
+        result.push_back(json::object_t::value_type(key, value));
+
+        /*BREAK HERE*/
+        //was this the last one? if yes, break. if no, resize queries and continue
+        if (nextQueryPos == queries.size())
+            break;
+
+        queries = queries.substr(nextQueryPos + 1);
+
+    } while (true);
+
+    return result.dump();
+}
+
+}
+
+namespace drivers
+{
+
+OneDrive::OneDrive(http::Pool& pool, std::unique_ptr<Auth> auth)
+    : Https(pool)
+    , m_auth(std::move(auth))
+{ }
+
+std::unique_ptr<OneDrive> OneDrive::create(http::Pool& pool, const std::string s)
+{
+    if (auto auth = Auth::create(s))
+    {
+        return makeUnique<OneDrive>(pool, std::move(auth));
+    }
+
+    return std::unique_ptr<OneDrive>();
+}
+
+std::unique_ptr<std::size_t> OneDrive::tryGetSize(const std::string path) const
+{
+    Resource resource(path);
+    http::Headers headers(m_auth->headers());
+    headers["Content-Type"] = "application/x-www-form-urlencoded";
+    drivers::Https https(m_pool);
+
+    const auto res(https.internalGet(resource.endpoint(), headers));
+
+    if (!res.ok())
+    {
+        std::cout <<
```

Comment:
lint

Reply:
This is how it was written in other parts of the code. https://github.com/connormanning/arbiter/blob/master/arbiter/drivers/google.cpp#L153-L154